src/__tests__/useA11yBlockLink.test.js

```diff
@@ -0,0 +1,144 @@
+import React, {useRef} from 'react';
+import {render, configure, fireEvent} from '@testing-library/react';
+import useA11yBlockLink from '../useA11yBlockLink';
+
+configure({testIdAttribute: 'data-test-id'});
+
+const Demo = ({testId, handleMainClick, handleThirdClick}) => {
+  const mainClickRef = useRef();
+  const secondRef = useRef();
+  const thirdRef = useRef();
+
+  const {handleClick: handleA11yClick} = useA11yBlockLink({
+    mainClick: mainClickRef,
+    secondRef,
+    thirdRef,
+  });
+
+  const handleClick = () => {
+    handleMainClick();
+    handleA11yClick();
+  };
+
+  return (
+    <div onClick={handleMainClick} data-test-id={`${testId}-wrapper`}>
+      <a is="a" ref={mainClickRef} data-test-id={`${testId}-main`} href="#main">
+        <img
+          data-test-id={`${testId}-image`}
+          src="https://dummyimage.com/300x300/000/fff"
+        />
+      </a>
+      <a href="#second" ref={secondRef} data-test-id={`${testId}-second`}>
+        Change Page
+      </a>
+      <p>Description text</p>
+      <button
+        type="button"
+        onClick={handleThirdClick}
+        ref={thirdRef}
+        data-test-id={`${testId}-third`}
+      >
+        Third Event
+      </button>
+    </div>
+  );
+};
+
+describe('useA11yBlockLink', () => {
+  global.window = Object.create(window);
+  const defineUrl = (url) => {
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: url,
+      },
+      writable: true,
+    });
+  };
+
+  it('fires main click on containing div', () => {
+    const mainClickMock = jest.fn();
+    const thirdClickMock = jest.fn();
+    const {container, queryByTestId} = render(
+      <Demo
+        testId="demo"
+        handleMainClick={mainClickMock}
+        handleThirdClick={thirdClickMock}
+      />
+    );
+
+    fireEvent.click(queryByTestId('demo-wrapper'));
+
+    // Expect click event to fire once
+    expect(mainClickMock).toHaveBeenCalledTimes(1);
+    // Expect the internal button was not clicked
+    expect(thirdClickMock).toHaveBeenCalledTimes(0);
+
+    // Manually update URL after click to window with the href
+    defineUrl(queryByTestId('demo-main').getAttribute('href'));
+
+    // Expect that firing the event will update the URL once manually updated
+    expect(window.location.href).toBe('#main');
```

Comment:
This seems a bit weird to me, almost like we're testing the test code itself and not the hook, e.g. this assertion is verifying out `defineUrl` thing is doing it's job, but its unrelated to the component being tested or the hook also.

Reply:
Updated to assert on a change in the href by using `location.assign`